LibSerialize.lua

```diff
@@ -1291,18 +1394,46 @@ function LibSerialize:SerializeEx(opts, ...)
     end
 
     self:_ClearReferences()
+
     return FlushWriter()
 end
 
+function LibSerialize:SerializeAsyncEx(callback, opts, ...)
+    if not CreateFrame then
+        error('Async is expected to be used within the WoW environment.')
+    end
+    opts.async = true
+    local dots = {...}
+    local processing = GetFrame()
+    local thread = coroutine_create(LibSerialize.SerializeEx)
+    processing:SetScript('OnUpdate', function()
+        if coroutine_status(thread) ~= 'dead' then
+            local success_serial, serialized = coroutine_resume(thread, LibSerialize, opts, unpack(dots))
+            if not success_serial then
+                callback(nil)
+            elseif serialized then
+                callback(serialized)
+            end
+        else
+            processing:SetScript('OnUpdate', nil)
+        end
+    end)
+end
+
+function LibSerialize:SerializeAsync(callback, ...)
+  return self:SerializeAsyncEx(callback, defaultOptions, ...)
+end
+
 function LibSerialize:Serialize(...)
     return self:SerializeEx(defaultOptions, ...)
 end
 
-function LibSerialize:DeserializeValue(input)
+function LibSerialize:DeserializeValue(input, opts)
     self:_ClearReferences()
     local ReadBytes, ReaderBytesLeft = CreateReader(input)
 
     self._readBytes = ReadBytes
+  opts = opts or {}
```

Comment:
Here and elsewhere - inconsistent indentation (you're using 2 instead of 4?)

Reply:
Do not modify function arguments. Make a copy